### PR TITLE
(WIP) (PUP-7042) Externalize warning and error strings

### PR DESCRIPTION
**Pull request by Magisus**
This PR tracks the work of marking warning and error strings for externalization and translation using gettext tooling.

### CODE REVIEW

File: lib/puppet/face/help.rb [hunk 1]
```diff
@@ -20,7 +21,7 @@
       $ puppet help
     EOT
 
-    option _("--version VERSION") do
+    option "--version" + _("VERSION") do
```
Comment (MosesMendoza) [inline lib/puppet/face/help.rb]: I think we lost a space here

File: lib/puppet/environments.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ module Puppet::Environments
   class EnvironmentNotFound < Puppet::Error
     def initialize(environment_name, original = nil)
       environmentpath = Puppet[:environmentpath]
-      super("Could not find a directory environment named '#{environment_name}' anywhere in the path: #{environmentpath}. Does the directory exist?", original)
+      super(_("Could not find a directory environment named '#{environment_name}' anywhere in the path: #{environmentpath}. Does the directory exist?", original))
```
Comment (MosesMendoza) [inline lib/puppet/environments.rb]: I think we want the first closing parens inside of `original` - that's not part of the string

**Comment by Magisus** [comment]
I will be adding commits to this as time goes on, but everything up here currently is ready for review. I will most likely be splitting this into many smaller PRs to make that review easier.

**Comment by Magisus** [comment]
This PR is superseded by the many individual PRs with a PUP-7042 ticket number. Keeping this branch open to help docs people review strings.

**Pull request closed without merging**